src/linearAlgebra/AbstractMatrix.h

```diff
@@ -40,6 +40,93 @@ namespace dftefe
     template <typename ValueType, dftefe::utils::MemorySpace memorySpace>
     class AbstractMatrix
     {
+    public:
+      /**
+       * @brief The value setter interface for the matrix class. This setter
+       *        assumes the given pointer to ValueType to be a serially repeated
+       *        matrix and copy the data to the corresponding local owned
+       *        location trivially.
+       * @warning There is no boundary check in this function. The user should
+       *          be responsible that the size of data array should have d_m*d_n
+       *          size.
+       * @param data The pointer to the data to be copied from.
+       */
+      virtual void
+      setValues(const ValueType *data) = 0;
+
+      /**
+       * @brief The value setter interface for the matrix class. This setter
+       *        assumes the given pointer to ValueType to be a serially repeated
+       *        matrix and copy the data to the corresponding local owned
+       *        sub-matrix (i1:i2-1, j1:j2-1) trivially.
+       * @warning There is no boundary check in this function. The user should
+       *          be responsible that the size of data array should have
+       *          (j2-j1)*(i2-i1) size.
+       * @param i1 the global index of the first row of the submatrix.
```

Comment:
Could you please clarify in the documentation whether the global index is counted starting from 0 or 1?